pkg/gitprovider/azure/azure.go

```diff
@@ -181,6 +181,79 @@ func (p *provider) ListPullRequests(
 	return pts, nil
 }
 
+// MergePullRequest implements gitprovider.Interface.
+func (p *provider) MergePullRequest(
+	ctx context.Context,
+	id int64,
+) (*gitprovider.PullRequest, bool, error) {
+	gitClient, err := adogit.NewClient(ctx, p.connection)
+	if err != nil {
+		return nil, false, fmt.Errorf("error creating Azure DevOps client: %w", err)
+	}
+
+	// Get the current PR to check its status and get the last merge source commit
+	currentPR, err := gitClient.GetPullRequest(ctx, adogit.GetPullRequestArgs{
+		Project:       &p.project,
+		RepositoryId:  &p.repo,
+		PullRequestId: ptr.To(int(id)),
+	})
+	if err != nil {
+		return nil, false, fmt.Errorf("error getting pull request %d: %w", id, err)
+	}
+	if currentPR == nil {
+		return nil, false, fmt.Errorf("pull request %d not found", id)
+	}
+
+	status := ptr.Deref(currentPR.Status, adogit.PullRequestStatusValues.NotSet)
+	mergeStatus := ptr.Deref(currentPR.MergeStatus, adogit.PullRequestAsyncStatusValues.NotSet)
+
+	switch status {
+	case adogit.PullRequestStatusValues.Completed:
+		pr, convertErr := convertADOPullRequest(currentPR)
+		if convertErr != nil {
+			return nil, false, fmt.Errorf("error converting pull request %d: %w", id, convertErr)
+		}
+		return pr, true, nil
+	case adogit.PullRequestStatusValues.Abandoned:
+		return nil, false, fmt.Errorf("pull request %d is abandoned", id)
+	case adogit.PullRequestStatusValues.Active:
+		// If merge status shows conflicts or policies not met → retry later
```

Comment:
I would encourage you to avoid talking about "retries."

This component doesn't know anything about retries.

All it does is tries to merge and indicates if it was successful or not. Any decisions about retries are made at a much higher level.